File: client/android/Data/Auth/Session/AuthSessionGateway.kt
package app.mobiling.client.data.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.ConfirmVerificationRequest
import app.mobiling.client.contract.auth.session.RegisterAuthRequest
import app.mobiling.client.contract.auth.session.StartAuthRequest
import app.mobiling.client.contract.auth.session.VerifySecondFactorRequest

interface AuthSessionGateway {
    suspend fun startAuth(request: StartAuthRequest): AuthSessionPayload
    suspend fun registerAuth(request: RegisterAuthRequest): AuthSessionPayload
    suspend fun restoreAuth(): AuthSessionPayload
    suspend fun logoutAuth()
    suspend fun resendVerification(): AuthSessionPayload
    suspend fun confirmVerification(request: ConfirmVerificationRequest): AuthSessionPayload
    suspend fun challengeSecondFactor(): AuthSessionPayload
    suspend fun verifySecondFactor(request: VerifySecondFactorRequest): AuthSessionPayload
}

File: client/ios/Data/Auth/Session/AuthSessionGateway.swift
public protocol AuthSessionGateway {
    func startAuth(request: StartAuthRequest) async throws -> AuthSessionPayload
    func registerAuth(request: RegisterAuthRequest) async throws -> AuthSessionPayload
    func restoreAuth() async throws -> AuthSessionPayload
    func logoutAuth() async throws
    func resendVerification() async throws -> AuthSessionPayload
    func confirmVerification(request: ConfirmVerificationRequest) async throws -> AuthSessionPayload
    func challengeSecondFactor() async throws -> AuthSessionPayload
    func verifySecondFactor(request: VerifySecondFactorRequest) async throws -> AuthSessionPayload
}